src/realm/sync/network/http.hpp

```diff
@@ -327,6 +339,37 @@ struct HTTPParser : protected HTTPParserBase {
             };
             m_socket.async_read(m_read_buffer.get(), *m_found_content_length, std::move(handler));
         }
+        else if (m_has_chunked_encoding) {
+            auto content_length_handler = [this](std::error_code ec, size_t chunk_start_index) {
+                if (ec == util::error::operation_aborted) {
+                    on_complete(ec);
+                    return;
+                }
+                auto content_length = hex_to_int(StringData(m_read_buffer.get(), chunk_start_index - 2));
+
+                if (content_length == 0) {
+                    on_body(m_chunked_encoding_ss->str());
+                    on_complete(ec);
+                    return;
+                }
+
+                auto handler = [this](std::error_code ec, size_t n) {
+                    if (ec == util::error::operation_aborted) {
+                        on_complete(ec);
+                        return;
+                    }
+                    auto chunk_data = StringData(m_read_buffer.get(), n - 2); // -2 to strip \r\n
```

Comment:
nit: I think within sync client code we're trying to keep it so `StringData` refers to string data inside a realm db and `std::string_view` is an unowned view of a sequence of strings.

Reply:
Addressed. There's still a bunch of code using `StringData` in `http.cpp` not related to this PR, I can remove all occurrences of `StringData` if you like.